### PR TITLE
Implement Caching

### PR DESCRIPTION
**Pull request by blkbee**
I used BigCache for admins because if the bot has many users, redis will slow down(latency). I also give redis as a second option(just change a bit code and it will use redis as the cacher).

### CODE REVIEW

**Comment by jumatberkah** [comment]
@soekarnohatta finally

File: go_bot/modules/muting/mute.go [hunk 1]
```diff
@@ -137,15 +137,7 @@ func tempMute(bot ext.Bot, u *gotgbot.Update, args []string) error {
 		return err
 	}
 
-	member, err := chat.GetMember(userId)
-	if err != nil {
-		if err.Error() == "User not found" {
-			_, err := msg.ReplyText("I can't seem to find this user!")
```
Comment (SphericalKat) [inline go_bot/modules/muting/mute.go:143]: Are you sure this sort of error will never occur?

File: go_bot/modules/utils/chat_status/chat_status.go [hunk 2]
```diff
@@ -51,20 +58,23 @@ func IsUserBanProtected(chat *ext.Chat, userId int, member *ext.ChatMember) bool
...
-		error_handling.HandleErr(err)
-		member = mem
+
+	admins, err := caching.CACHE.Get(fmt.Sprintf("admin_%v", chat.Id))
+	if err != nil {
+		admincache(chat)
 	}
-	if member.Status == "administrator" || member.Status == "creator" {
+
+	var x Cache
```
Comment (SphericalKat) [inline go_bot/modules/utils/chat_status/chat_status.go]: I'd love it if you could change these var names from xyz to the first letter of the type they represent :)

File: go_bot/modules/utils/chat_status/chat_status.go [hunk 3]
```diff
@@ -156,3 +166,18 @@ func contains(s []string, e string) bool {
 	}
 	return false
 }
+
+func admincache(chat *ext.Chat) {
+	x, err := chat.GetAdministrators()
```
Comment (SphericalKat) [inline go_bot/modules/utils/chat_status/chat_status.go]: Same variable naming scheme applies here

**Review by SphericalKat — CHANGES_REQUESTED** [review]
Overall looks good, please make the requested changes and I'll merge this

File: go_bot/modules/utils/chat_status/chat_status.go (hunk 2 above)
Comment (zenoember) [reply go_bot/modules/utils/chat_status/chat_status.go]: Sure!

File: go_bot/modules/utils/chat_status/chat_status.go (hunk 3 above)
Comment (zenoember) [reply go_bot/modules/utils/chat_status/chat_status.go]: Ok, i'll change it

**Review by SphericalKat — APPROVED** [review]
Approved, merging.

**Pull request merged**